protoc-gen-swagger/genswagger/atlas_patch.go

```diff
@@ -157,12 +157,30 @@ The service-defined string used to identify a page of resources. A null value in
 						schema.Properties = map[string]spec.Schema{}
 					}
 
-					schema.Properties["success"] = *(&spec.Schema{}).WithProperties(
-						map[string]spec.Schema{
-							"status":  *spec.StringProperty().WithExample(opToStatus(on)),
-							"code":    *spec.Int32Property().WithExample(opToCode(on)),
-							"message": *spec.StringProperty().WithExample("<response message>"),
-						})
+					successSchema := map[string]spec.Schema{
+						"status":  *spec.StringProperty().WithExample(opToStatus(on)),
+						"code":    *spec.Int32Property().WithExample(opToCode(on)),
+						"message": *spec.StringProperty().WithExample("<response message>"),
+					}
+
+					def := sw.Definitions[trim(rsp.Schema.Ref)]
+					for fn, v := range def.Properties {
+						if v.Ref.String() == "#/definitions/apiPageInfo" {
+							successSchema["_offset"] = *spec.Int32Property().WithExample(5).
+								WithDescription("The service may optionally include the offset of the next page of resources. A null value indicates no more pages.")
+
+							successSchema["_size"] = *spec.StringProperty().WithExample(25).
+								WithDescription("The service may optionally include the total number of resources being paged.")
+
+							successSchema["_pagetoken"] = *spec.StringProperty().WithExample(opToStatus(on)).
+								WithDescription("The service response may optionally contain a string to indicate the next page of resources. A null value indicates no more pages.")
+							delete(def.Properties, fn)
+						}
```

Comment:
break

Reply:
or should we be aware of having multiple pageInfo inside message?

Reply:
we need to double check how this is handled by gateway. I think it handles only the first one.

Reply:
successSchema is a map and we have unique keys such as _offset, _size and_pagetoken, i think duplication is impossible, but if we break and we add two field page info, the second one will not be deleted, because in loop we delete field typed as PageInfo

Reply:
this will not harm the successSchema, however delete(def.Properties, fn) could miss second property that refers to apiPageInfo

Reply:
anyway, it doesn't matter, if one adds second page info we should left it as a flaw in schema.

Reply:
Okey, I add break statement if we find field apiPageInfo, review again please